pkg/services/accesscontrol/resourceservices/resource_services.go

```diff
@@ -0,0 +1,93 @@
+package resourceservices
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/accesscontrol/resourcepermissions"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+func ProvideResourceServices(router routing.RouteRegister, sql *sqlstore.SQLStore, ac accesscontrol.AccessControl, store accesscontrol.ResourcePermissionsStore) (*ResourceServices, error) {
+	teamPermissions, err := ProvideTeamPermissions(router, sql, ac, store)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ResourceServices{services: map[string]*resourcepermissions.Service{
+		"teams": teamPermissions,
+	}}, nil
+}
+
+type ResourceServices struct {
+	services map[string]*resourcepermissions.Service
+}
+
+func (s *ResourceServices) GetTeamService() *resourcepermissions.Service {
+	return s.services["teams"]
+}
+
+var (
+	TeamMemberActions = []string{
+		"teams:read",
+	}
+
+	TeamAdminActions = []string{
+		"teams:read",
+		"teams:delete",
+		"teams:write",
+		"teams.permissions:read",
+		"teams.permissions:write",
+	}
+)
+
+func ProvideTeamPermissions(router routing.RouteRegister, sql *sqlstore.SQLStore, ac accesscontrol.AccessControl, store accesscontrol.ResourcePermissionsStore) (*resourcepermissions.Service, error) {
+	options := resourcepermissions.Options{
+		Resource:    "teams",
+		OnlyManaged: true,
+		ResourceValidator: func(ctx context.Context, orgID int64, resourceID string) error {
```

Comment:
nit: given that we have a resource service abstraction tests, we can skip testing simple units here. However something to consider as the logic can get complicated.